### PR TITLE
Add guide on debugging memory issues

### PR DESCRIPTION
**Pull request by justindeguzman**

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit 4c4d6a3fc1702ca36cc99aafcd7e235ce3a983ad with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/1534/4c4d6a3fc1702ca36cc99aafcd7e235ce3a983ad/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs Check</td><td>Builds and tests the documentation</td><td>✅ success</td></tr>
<tr><td>Vercel</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: docs/en/guides/developer/debugging-memory-issues.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+---
```
Comment (nickitat) [inline docs/en/guides/developer/debugging-memory-issues.md:1]: I think it should be a KB article. wdyt?

File: docs/en/guides/developer/debugging-memory-issues.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+slug: /en/guides/developer/debugging-memory-issues
+sidebar_label: Debugging Memory Issues
+sidebar_position: 1
+description: Queries to help you debug memory issues.
+---
+
+# Debugging memory issues
+
+When encountering memory issues, it is helpful to know what queries and resources are consuming a significant amount of memory. Below are queries that can help find which queries, databases, and tables can be optimized:
+
```
Comment (nickitat) [inline docs/en/guides/developer/debugging-memory-issues.md:11]: pls add also:

``` sql
SELECT formatReadableSize(sum(bytes_allocated)) FROM system.dictionaries;
```

``` sql
SELECT
    sumIf(data_uncompressed_bytes, part_type = 'InMemory') as memory_parts,
    formatReadableSize(sum(primary_key_bytes_in_memory)) AS primary_key_bytes_in_memory,
    formatReadableSize(sum(primary_key_bytes_in_memory_allocated)) AS primary_key_bytes_in_memory_allocated
FROM system.parts;
```

File: docs/en/guides/developer/debugging-memory-issues.md (hunk 1 above)
Comment (justindeguzman) [reply docs/en/guides/developer/debugging-memory-issues.md:1]: I think we should start having "official guides" vs. relying on KB articles.

In my opinion, KB articles should be predominantly used for Q&A-format articles.

File: docs/en/guides/developer/debugging-memory-issues.md (hunk 2 above)
Comment (justindeguzman) [reply docs/en/guides/developer/debugging-memory-issues.md:11]: Added! Please let me know if their descriptions are correct:

- **Output total memory used by dictionaries**
- **Output total memory used by primary keys**

File: docs/en/guides/developer/debugging-memory-issues.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+**List queries by peak memory usage**
```
Comment (nickitat) [inline docs/en/guides/developer/debugging-memory-issues.md:12]:
```suggestion
**List currently running processes by peak memory usage**
```

File: docs/en/guides/developer/debugging-memory-issues.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    query,
+    elapsed,
+    formatReadableSize(memory_usage),
+    formatReadableSize(peak_memory_usage),
+FROM system.processes
+ORDER BY peak_memory_usage DESC
+LIMIT 100;
+```
+
+**List metrics based on total memory usage**
```
Comment (nickitat) [inline docs/en/guides/developer/debugging-memory-issues.md:26]:
```suggestion
**List metrics of memory usage**
```

File: docs/en/guides/developer/debugging-memory-issues.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+FROM
+    system.asynchronous_metrics
+WHERE
+    metric like '%Cach%'
+    or metric like '%Mem%'
+order by
+    value desc;
+```
+
+**List databases by current memory usage**
```
Comment (nickitat) [inline docs/en/guides/developer/debugging-memory-issues.md:40]:
```suggestion
**List tables by current memory usage**
```

File: docs/en/guides/developer/debugging-memory-issues.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+```sql
+SELECT
+    database,
+    name,
+    formatReadableSize(total_bytes)
+FROM system.tables
+WHERE engine IN ('Memory','Set','Join');
+```
+
+**List merges by current memory usage**
```
Comment (nickitat) [inline docs/en/guides/developer/debugging-memory-issues.md:51]:
```suggestion
**Output total memory used by merges**
```

File: docs/en/guides/developer/debugging-memory-issues.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+```sql
+SELECT formatReadableSize(sum(memory_usage)) FROM system.merges;
+```
+
+**List processes by current memory usage**
```
Comment (nickitat) [inline docs/en/guides/developer/debugging-memory-issues.md:57]:
```suggestion
**Output total memory used by currently running processes**
```

File: docs/en/guides/developer/debugging-memory-issues.md (hunk 1 above)
Comment (nickitat) [reply docs/en/guides/developer/debugging-memory-issues.md:1]: this section is for developer guides. this article is supposed to be used by all CH users.

**Pull request merged**